sql: add statement_id to crdb_internal.node_statement_statistics

Azhng opened the pull request:
Currently, crdb_internal.node_transaction_statistics uses the
statement_ids column to reference statements in
crdb_internal.node_statement_statistics. However, the statement
statistics view does not have column that shows statement id.

This commit introduce a new statement_id column in the statement
statistics view.

Release note (sql change): crdb_internal.node_statement_statistics
 now stores statement_id.

Closes: #64072

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/64076)
<!-- Reviewable:end -->

arulajmani reviewed (commented):
Talked offline, and we agreed that going through the STRING -> INT switch at this point isn't worth it considering the backwards compat issue. Other than that, things on here look good. Nice find on #64072! 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/64076#-:-MYuzSrEEvvCcj0jwrqd:b-v83180)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Azhng committed:
sql: add statement_id to crdb_internal.node_statement_statistics

Currently, crdb_internal.node_transaction_statistics uses the
statement_ids column to reference statements in
crdb_internal.node_statement_statistics. However, the statement
statistics view does not have column that shows statement id.

This commit introduce a new statement_id column in the statement
statistics view.

Release note (sql change): crdb_internal.node_statement_statistics
 now stores statement_id.

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/64076#-:-MYvRn0w82bH_e8PBsFp:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

Azhng commented:
TFTR!

bors r=arulajmani
